### ISSUE TITLE
Create Gradle Task to Run GatorGrader

### ISSUE THREAD

**Issue by Michionlion** [gkapfham/gatorgrader#14]
Currently students run a bash file which calls the gatorgrader python program to do checking. Once @toccinAC finishes documentation and converting that bash file setup to a python one, a custom gradle task that runs gatorgrader for a project would greatly help students. We should put together a JAR or other packaged gradle add-on to allow a command such as `$gradle grade` to be used.

**Comment by Michionlion** [comment]
As a related issue, we should define a format for a file that assignment requirements should be made in - then the actual checking can be separated from the command-line arguments to gatorgrader, and @toccinAC's python program could parse that file and then call `gatorgrader.py` (I guess we could call that 'gatorgrader core' with the necessary arguments.

**Comment by gkapfham** [comment]
Hello @Michionlion! I agree that this is a really good idea. Can you start to investigate how to perform this type of packing in Gradle? Perhaps you might be able to learn more about how to allow Gradle tasks to call Python programs in advance of @toccinAC completing the new Python script? Then, once he is finished, it would be possible to connect the new script to your Gradle approach. Seems reasonable?

**Comment by Michionlion** [comment]
I've started work on this feature, in the form of a plugin to be published on one of the central Gradle repositories. Anyone is invited to join - it is written in Java with the Gradle API. The repository is [here](https://github.com/Michionlion/gatorgrader-plugin).

**Comment by Michionlion** [comment]
The first working build of GatorGradle has been released. It's available at the [gradle plugin portal](https://plugins.gradle.org/plugin/org.gatored.gatorgradle), and can be included from any gradle project to enable GatorGrading checking. The documentation and future work (still need major improvements) can be found at the [gatored/gatorgradle repo](https://github.com/gatored/gatorgradle).

I'm now closing this issue -- any future discussion should take place on the projects and issues board in the above repository.
